### DISCUSSION TITLE
Enhance suggestions with local pending Git changes

### DISCUSSION THREAD

**Discussion by paulirwin** [community/community#118437] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

Forgive me if this has been discussed before; I did a search and didn't find anything that was a close match. I am working in a repo that has projects in a couple different languages including C# and TypeScript. I added a feature flag name in the C# code in Rider with Copilot, then went to do a similar change to the TypeScript code in VS Code, and it didn't pick up on the suggestion. That C# code change was a pending change in my local git repo, so Copilot could have looked at that and added it to its RAG as something that I likely might do, even though it was in a different language and IDE. 

This also could be useful for i.e. when you add to a Markdown file in the repo, that pending change could be "top of mind" for the LLM to suggest code in other languages/editors for the same repo.

It also would be nice if it could consider code from git commits in the current branch that are different than main/master, as those are likely to be relevant to subsequent commits on that branch.

**Comment by RubenGamarrarodriguez-tomtom** [comment]
🤔Maybe not what you are looking for but that way of working might not be the best in typical projects. If there is advantages with mixing languages like C# and TypeScript in the current branch in some scenarios, it's fine. But probably most of the cases would benefit from writing small prs when possible and when the changes can be done in small increments https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/getting-started/helping-others-review-your-changes#write-small-prs or even some of the parts of the role description at https://www.feature23.com/careers/software-engineer
